qrenderdoc/Widgets/Extended/RDSplitter.cpp

```diff
@@ -0,0 +1,137 @@
+#include "RDSplitter.h"
+
+#include <QPainter>
+#include <QPaintEvent>
+#include <QApplication>
+
+
+RDSplitterDoubleClickEvent::RDSplitterDoubleClickEvent(int index)
+    :QEvent(staticType()), m_index(index)
+{
+
+}
+
+int RDSplitterDoubleClickEvent::index() const
+{
+    return m_index;
+}
+
+QEvent::Type RDSplitterDoubleClickEvent::staticType()
+{
+    static int type = registerEventType();
+    return (QEvent::Type)type;
+}
+
+RDSplitterHandle::RDSplitterHandle(Qt::Orientation orientation, QSplitter * parent)
+    :QSplitterHandle(orientation, parent), m_index(-1), m_isCollapsed(false)
+{
+
+}
+
+void RDSplitterHandle::setIndex(int index)
+{
+    m_index = index;
+}
+
+int RDSplitterHandle::index() const
+{
+    return m_index;
+}
+
+void RDSplitterHandle::setTitle(const QString &title)
+{
+    m_title = title;
+}
+
+const QString &RDSplitterHandle::title() const
+{
+    return m_title;
+}
+
+void RDSplitterHandle::setCollapsed(bool collapsed)
+{
+    m_isCollapsed = collapsed;
+}
+
+bool RDSplitterHandle::collapsed() const
+{
+    return m_isCollapsed;
+}
+
+void RDSplitterHandle::paintEvent(QPaintEvent * event)
+{
+    QPainter painter(this);
+    painter.setPen(Qt::black);
+
+    QString text(m_title);
+    if (m_isCollapsed) text.append(" ...");
+
+    painter.drawText(event->rect(), Qt::AlignCenter | Qt::TextWordWrap, text);
+}
+
+void RDSplitterHandle::mouseDoubleClickEvent(QMouseEvent *event)
+{
+    // post the event to the parent
+    RDSplitterDoubleClickEvent *rdEvent = new RDSplitterDoubleClickEvent(m_index);
+    QApplication::postEvent(parent(), rdEvent);
```

Comment:
I don't think the added complexity of the `RDSplitterDoubleClickEvent` is worthwhile.

We know that a `RDSplitterHandle` will be used with a `RDSplitter`, so when you construct the `RDSplitterHandle` and the `QSplitter*` is passed in, just cast it to `RDSplitter*` and call a 'handleDoubleClicked(index)' member function directly.

If you wanted to avoid making such coupled assumptions and still be Qt-like, you could make this instead a signal `doubleClicked(index)` that you `emit` here and then inside `RDSplitter::createHandle` you could connect the signal to the `handleDoubleClicked(index)` function as a slot.